rust/main/agents/relayer/src/relayer.rs

```diff
@@ -419,10 +419,13 @@ impl Relayer {
     ) -> Instrumented<JoinHandle<()>> {
         let index_settings = self.as_ref().settings.chains[origin.name()].index_settings();
         let contract_sync = self.message_syncs.get(origin).unwrap().clone();
-        let cursor = contract_sync
-            .cursor(index_settings)
-            .await
-            .unwrap_or_else(|err| panic!("Error getting cursor for origin {origin}: {err}"));
+        let cursor = match contract_sync.cursor(index_settings).await {
+            Ok(cursor) => cursor,
+            Err(err) => {
+                error!(?err, origin=?origin, "Error building cursor for origin");
```

Comment:
I wonder if we need a high urgency alert once such error is logged?

Reply:
added a new `critical_error` metric per chain that we'll alert on